common/src/main/java/io/pravega/common/util/Retry.java

```diff
@@ -217,21 +218,22 @@ private RetryAndThrowBase(Class<ThrowsT> throwType, RetryWithBackoff params) {
         public CompletableFuture<Void> runInExecutor(final Runnable task,
                                                      final ScheduledExecutorService executorService) {
             Preconditions.checkNotNull(task);
-            CompletableFuture<Void> result = new CompletableFuture<>();
+            AtomicBoolean isDone = new AtomicBoolean();
             AtomicInteger attemptNumber = new AtomicInteger(1);
             AtomicLong delay = new AtomicLong(0);
-            FutureHelpers.loop(
-                    () -> !result.isDone(),
+            return FutureHelpers.loop(
+                    () -> !isDone.get(),
                     () -> FutureHelpers.delayedFuture(Duration.ofMillis(delay.get()), executorService)
                             .thenRunAsync(task, executorService)
-                            .thenAccept(result::complete) // We are done.
+                            .thenAccept(v -> isDone.set(true)) // We are done.
```

Comment:
you can use `thenRun` (since you don't care about the return value of the previous call)

Reply:
done.